conans/model/graph_lock.py

```diff
@@ -2,297 +2,465 @@
+        self._path = path
+        if not revisions_enabled:
+            if ref:
+                self._ref = ref.copy_clear_rev()
+            if prev:
+                self._prev = DEFAULT_REVISION_V1
+
+    @property
+    def requires(self):
+        return self._requires
+
+    @property
+    def build_requires(self):
+        return self._build_requires
+
+    def relax(self):
+        self._relaxed = True
+
+    def clean_modified(self):
+        self._modified = None
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def ref(self):
+        return self._ref
+
+    @property
+    def python_requires(self):
+        return self._python_requires
+
+    @ref.setter
+    def ref(self, value):
+        # only used at export time, to assign rrev
+        if not self._revisions_enabled:
+            value = value.copy_clear_rev()
+        if self._ref:
+            if (self._ref.copy_clear_rev() != value.copy_clear_rev() or
+                    (self._ref.revision and self._ref.revision != value.revision)):
+                raise ConanException("Attempt to modify locked %s to %s"
+                                     % (repr(self._ref), repr(value)))
+
+        # Just in case
+        self._ref = value
+        self._package_id = None
+        self._prev = None
+        self._path = None
+
+    @property
+    def package_id(self):
+        return self._package_id
+
+    @package_id.setter
+    def package_id(self, value):
+        if (self._package_id is not None and self._package_id != PACKAGE_ID_UNKNOWN and
+                self._package_id != value):
+            raise ConanException("Attempt to change package_id of locked '%s'" % repr(self._ref))
+        if value != self._package_id:  # When the package_id is being assigned, prev becomes invalid
+            self._prev = None
+        self._package_id = value
+
+    @property
+    def prev(self):
+        return self._prev
+
+    @prev.setter
+    def prev(self, value):
+        if not self._revisions_enabled and value is not None:
+            value = DEFAULT_REVISION_V1
+        if self._prev is not None:
+            raise ConanException("Trying to modify locked package {}".format(repr(self._ref)))
+        if self._prev != value:
+            self._modified = True  # Only for conan_build_info
+        self._prev = value
+
+    @property
+    def options(self):
+        return self._options
+
+    def only_recipe(self):
+        self._package_id = None
+        self._prev = None
+        self._options = None
+        self._modified = None
 
     @staticmethod
-    def from_dict(data):
+    def deserialize(data, revisions_enabled):
         """ constructs a GraphLockNode from a json like dict
         """
-        json_pref = data["pref"]
-        pref = PackageReference.loads(json_pref, validate=False) if json_pref else None
+        json_ref = data.get("ref")
+        ref = ConanFileReference.loads(json_ref) if json_ref else None
```

Comment:
No need to validate if it was stored in a lockfile

```suggestion
        ref = ConanFileReference.loads(json_ref, validate=False) if json_ref else None
```